pkg/app/kafka.go

```diff
@@ -260,36 +282,79 @@ func (kc *KafkaCluster) Initialize(ctx context.Context) error {
 	return nil
 }
 
-// to produce messages to kafka
-func produce(ctx context.Context, topic string, namespace string) error {
-	partition := 0
-	forwarder, err := K8SServicePortForward(ctx, bootstrapServerHost, namespace, bootstrapServerPort)
+type Message struct {
+	Topic     string `json:"topic"`
+	Key       string `json:"key"`
+	Value     string `json:"value"`
+	Partition int    `json:"partition"`
+	Offset    int    `json:"offset"`
+}
+
+type InsertPayload struct {
+	Records []Records `json:"records"`
+}
+
+type Records struct {
+	Key   string `json:"key"`
+	Value string `json:"value"`
+}
+
+type Payload struct {
+	Name                     string `json:"name"`
+	AutoOffsetReset          string `json:"auto.offset.reset"`
+	Format                   string `json:"format"`
+	EnableAutoCommit         bool   `json:"enable.auto.commit"`
+	FetchMinBytes            int    `json:"fetch.min.bytes"`
+	ConsumerRequestTimeoutMs int    `json:"consumer.request.timeout.ms"`
+}
+
+type SubscriptionPayload struct {
+	Topics []string `json:"topics"`
+}
+
+// InsertRecord creates a topic and insert a record
+func InsertRecord(ctx context.Context, namespace string) error {
+	forwarder, err := K8SServicePortForward(ctx, bridgeServiceName, namespace, bridgeServicePort)
 	if err != nil {
 		return err
 	}
-
 	defer forwarder.Close()
 	ports, err := forwarder.GetPorts()
 	if err != nil {
 		return err
 	}
-	uri := "localhost:" + fmt.Sprint(ports[0].Local)
+	uri := "http://localhost:" + fmt.Sprint(ports[0].Local)
 
-	conn, err := kafka.DialLeader(ctx, "tcp", uri, topic, partition)
+	data := InsertPayload{
+		Records: []Records{
+			{
+				Value: "sales-lead-0001",
+			},
+		},
+	}
+	payloadBytes, err := json.Marshal(data)
 	if err != nil {
 		return err
 	}
+	body := bytes.NewReader(payloadBytes)
 
-	defer conn.Close()
+	req, err := http.NewRequest("POST", uri+"/topics/"+topic, body)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/vnd.kafka.json.v2+json")
 
-	err = conn.SetWriteDeadline(time.Now().Add(10 * time.Second))
+	resp, err := http.DefaultClient.Do(req)
```

Comment:
We should also check for the correct response code. It will not raise an error when the response code is not 200/201